packages/core/src/contrib/ConstraintsUtils.ts

```diff
@@ -192,6 +193,42 @@ export const overlappingCircleLine = (
   return sub(d, add(r, o));
 };
 
+/**
+ * Require that circle `s1` overlaps line `s2` with some overlap `overlap`.
+ */
+export const overlappingTextLine = (
```

Comment:
lmao I think I need a Penrose diagram to help me understand how this function works

Reply:
Mostly copied from the signed distance functions by @joshsunshine :p. Still blocked on this until the Minkowski issue with text width/height is resolved